### ISSUE TITLE
No success trying to use a custom font

### ISSUE THREAD

**Issue by JerryBels** [KnpLabs/snappy#209]
Hello,

I tried everything I know to have my PDF displaying with Open Sans. I added it with CDN at first : 

```
<link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700,600' rel='stylesheet' type='text/css'>
```

Then tried in a local file : 

```
@font-face {
        font-family: "Open Sans";
        src: url("http://myserver.com/OpenSans-Regular.ttf") format('truetype');
    }
```

Then tried with base64 encoded font... When opening the HTML page all is right, but when I try to generate the PDF, by `getOutput` or by `generate`, it simply doesn't work, the font is never the wanted Open Sans.

Thanks ahead for any help !

**Comment by MitchellBouwman** [comment]
You can just use Google fonts import css:

`@import 'https://fonts.googleapis.com/css?family=Indie+Flower';`
`.IndieFlower{font-size: 25px; font-family: 'Indie Flower', cursive;}`

**Comment by benjaminprojas** [comment]
I am also having this issue, however I have local font files to use.  Is there a suggestion for getting that to work? When I check which fonts are embedded, its a completely different font than What I am setting via CSS.

**Comment by MitchellBouwman** [comment]
@benjaminprojas Same as mentioned above.

**Comment by beany-vu** [comment]
@MitchellBouwman  I tried your solution but it didn't work. :(

**Comment by MitchellBouwman** [comment]
@beany-vu You could try also implement the 
`<link href="https://fonts.googleapis.com/css?family=Indie+Flower" rel="stylesheet">`
In the <head> of your document, did you try with both?

I'm currently loading in 4 custom (google) fonts and it seems to work fine for me (without the <link href).

Also I have the CSS imported as following:
```
$css = file_get_contents("https://www.picbox.nl/static/css/pdfengine.css");

                        $html = "

                            <html>
                                <head>
                                    <style>
                                        $css
                                    </style>
                                    <meta charset=\"UTF-8\" />
                                </head>
                                <body>
                                ...
                                </body>

```

Where $css is in the <head><style>XX</style></head>

**Comment by NiR-** [comment]
As said by @MitchellBouwman following html works well :

```html
<!DOCTYPE html>
<html>
  <head>
    <style>
      @import 'https://fonts.googleapis.com/css?family=Indie+Flower';
      .IndieFlower{font-size: 25px; font-family: 'Indie Flower', cursive;}
      .cursive{font-family: cursive;}
    </style>
  </head>
  <body>
    <p class="IndieFlower">Lorem ipsum</p>
    <p class="cursive">Lorem ipsum</p>
  </body>
</html>
```

![image](https://user-images.githubusercontent.com/557933/30368294-0a9f2a0c-9871-11e7-877f-67d41afdf37d.png)

**Comment by NiR-** [comment]
That's really strange... Which version of wkhtmltopdf do you use?

**Comment by beany-vu** [comment]
I use 0.11.0-RC1

**Comment by NiR-** [comment]
Snappy requires wkhtmltopdf:~0.12, partly because older versions are too buggy ;) You should have no problem with latest version. Based on that, can I close this issue?

**Comment by beany-vu** [comment]
It's okay to me. Thanks!!! 👯‍♂️

**Comment by Ardakilic** [comment]
Necro bump, this was what I did to embed a custom font a couple of months ago

```
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Title</title>
<style type="text/css">
@font-face {
	font-family: 'myFontName';
	src: url(data:font/truetype;charset=utf-8;base64,base 64 encoded font stuff here) format('truetype');
	font-weight: 500;
	font-style: normal;
}
</style>
....
```

There are multiple ways to convert the fonts to base64 encoded strings. You can easily convert and embed to your pdf directly.

**Comment by Ardakilic** [comment]
@zemiacsik you either need to upload to web and source it like the above examples, or embed the font into pdf directly as I've stated. It's possibly looking for the font in the target's machine, which is not available.

**Comment by zemiacsik** [comment]
@Ardakilic thank you, the problem was, as usual, somewhere completely else.. it appeared that there is a css file attached by default with these old routes too 🤦‍♂️

**Comment by timching** [comment]
Hi all, I am encountering a similar problem too. But in our Symfony project, the pdf may contain different languages, so embedding fonts by using base64 may not be suitable for us.

we are using ubuntu 18 /w wkhtmltopdf 0.12.4, and I tried to use `<link>` tag to load Noto fonts by page language dynamically, but it didn't work. and I tried to `get {url}` & use the wkhtmltopdf command and convert it to pdf, the string character is correct(verify by using `strings output.pdf } | grep FontName`), so I don't think it is wkhtmltopdf binary problem.

did anyone know is there any other solution(s) (no need to use base64) that can fix the character issue?

Thank you!
